packages/cli/src/modules/chat-hub/chat-hub.attachment.service.ts

```diff
@@ -0,0 +1,154 @@
+	private readonly maxTotalSizeBytes = 200 * 1024 * 1024; // 200 MB
+
+	constructor(
+		private readonly binaryDataService: BinaryDataService,
+		private readonly messageRepository: ChatHubMessageRepository,
+	) {}
+
+	/**
+	 * Stores attachments through BinaryDataService.
+	 * This populates the 'id' field for attachments. When external storage is used,
+	 * BinaryDataService replaces base64 data with the storage mode string (e.g., "filesystem-v2").
+	 */
+	async store(
+		sessionId: ChatSessionId,
+		messageId: ChatMessageId,
+		attachments: IBinaryData[],
+	): Promise<IBinaryData[]> {
+		let totalSize = 0;
+		const attachmentsWithBuffer: Array<[IBinaryData, Buffer<ArrayBuffer>]> = [];
+
+		for (const attachment of attachments) {
+			if (attachment.id || !attachment.data) {
+				continue;
+			}
+
+			const buffer = Buffer.from(attachment.data, BINARY_ENCODING);
+			totalSize += buffer.length;
+
+			if (totalSize > this.maxTotalSizeBytes) {
+				const maxSizeMB = Math.floor(this.maxTotalSizeBytes / (1024 * 1024));
+
+				throw new BadRequestError(
+					`Total size of attachments exceeds maximum size of ${maxSizeMB} MB`,
+				);
+			}
+
+			if (attachment.fileName) {
+				attachment.fileName = sanitizeFilename(attachment.fileName);
+			}
+
+			attachmentsWithBuffer.push([attachment, buffer]);
+		}
+
+		return await Promise.all(
+			attachmentsWithBuffer.map(
+				async ([attachment, buffer]) =>
+					await this.binaryDataService.store(
+						{ type: 'chat-hub-message-attachment', sessionId, messageId },
+						buffer,
+						attachment,
+					),
+			),
+		);
+	}
+
+	/*
+	 * Gets a specific attachment from a message by index and returns it as either buffer or stream
+	 */
+	async getAttachment(
+		sessionId: ChatSessionId,
+		messageId: ChatMessageId,
+		attachmentIndex: number,
+	): Promise<
+		[
+			IBinaryData,
+			(
+				| { type: 'buffer'; buffer: Buffer<ArrayBufferLike>; fileSize: number }
+				| { type: 'stream'; stream: Stream.Readable; fileSize: number }
+			),
+		]
+	> {
+		const message = await this.messageRepository.getOneById(messageId, sessionId, []);
+
+		if (!message) {
+			throw new NotFoundError('Message not found');
+		}
+
+		const attachment = message.attachments?.[attachmentIndex];
+
+		if (!attachment) {
+			throw new NotFoundError('Attachment not found');
+		}
+
+		if (attachment.id) {
+			const metadata = await this.binaryDataService.getMetadata(attachment.id);
+			const stream = await this.binaryDataService.getAsStream(attachment.id);
+
+			return [attachment, { type: 'stream', stream, fileSize: metadata.fileSize }];
+		}
+
+		if (attachment.data) {
+			const buffer = await this.binaryDataService.getAsBuffer(attachment);
+
+			return [attachment, { type: 'buffer', buffer, fileSize: buffer.length }];
+		}
+
+		throw new NotFoundError('Attachment has no stored file');
+	}
+
+	/**
+	 * Deletes all files attached to messages in the session
+	 */
+	async deleteAllBySessionId(sessionId: string): Promise<void> {
```

Comment:
I think these ever get deleted only if user deletes them from the UI, right? This is fine for now but you need to eventually come up with a pruning strategy as we can't allow unlimited data growth eg on cloud